octavia-cli/octavia_cli/api_http_headers.py

```diff
@@ -25,7 +26,7 @@ class ApiHttpHeader:
     def __post_init__(self):
         try:
             assert isinstance(self.name, str) and self.name
-            assert isinstance(self.value, str) and self.value
+            assert (isinstance(self.value, str) or isinstance(self.value, MagicMock)) and self.value
```

Comment:
What's the right thing to do here?  Is importing a library only used for testing into the main codepath OK?

Reply:
`__post_init__` role is to validate user inputs (to check they are passing non-empty strings as header name and value. I'd prefer you monkey patch `__post_init__` in the test instead of allowing mock objects to be passed.

Reply:
You made these change to make the unit test pass? Let me know if you need help patching stuff :)

Reply:
Ok, I think I've got it! 17f07f9ddb61a481e15af8a1a99b8f252a3e72b8